tests/components/sense/test_sensor.py

```diff
@@ -0,0 +1,267 @@
+"""The tests for Sense sensor platform."""
+
+from datetime import timedelta
+from unittest.mock import MagicMock, PropertyMock
+
+from homeassistant.components.sense.const import (
+    ACTIVE_UPDATE_RATE,
+    CONSUMPTION_ID,
+    CONSUMPTION_NAME,
+)
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN, SensorDeviceClass
+from homeassistant.const import (
+    ATTR_DEVICE_CLASS,
+    ATTR_FRIENDLY_NAME,
+    ATTR_ICON,
+    STATE_UNAVAILABLE,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+from homeassistant.util.dt import utcnow
+
+from .conftest import (
+    DEVICE_1_ICON,
+    DEVICE_1_ID,
+    DEVICE_1_NAME,
+    DEVICE_1_POWER,
+    DEVICE_2_ICON,
+    DEVICE_2_ID,
+    DEVICE_2_NAME,
+    DEVICE_2_POWER,
+    MONITOR_ID,
+    setup_platform,
+)
+
+from tests.common import async_fire_time_changed
+
+
+async def test_device_power_sensors(
+    hass: HomeAssistant, entity_registry: er.EntityRegistry, mock_sense: MagicMock
+) -> None:
+    """Test the Sense device power sensors."""
+    await setup_platform(hass, SENSOR_DOMAIN)
+
+    entity = entity_registry.async_get(
```

Comment:
To avoid all these assertions, it might be easier to look at `snapshot_platform` helper:
`from tests.common import snapshot_platform`